### PR TITLE
Adding some initial thoughts regarding the Dev Team Charter

### PR DESCRIPTION
**Pull request by fedro321**
First stab at the Dev Team Charter. Establishing peer review and merging workflow also.

### CODE REVIEW

**Review by IzzieKi — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -1,2 +1,24 @@
 # DevTeamCharter
 
+## Mission
+The Dev Team are here to solve the problems of the Scottish Tech Army by delivering usable software. In turn this will help people during the current crisis and recovery brought on by the Covid-19 outbreak.
+
+## The Team
+The team is made up of skilled technologists, for the most part software developers with the capability to contribute to applications that will be deployed on the internet. An individuals value to the team will be measured primarily by their contribution.
```
Comment (alankennedy14) [inline README.md:7]: The team is made up of a mixture of enthusiastic volunteers with a mix of different skilled technologists, for the most part software developers with the capability to contribute to applications that will be deployed on the internet. An individuals value to the team will be measured primarily by their contribution.

**Review by LiamLamb — COMMENTED** [review]
Sounds all good to me :)

What Git strategies are we wanting to adopt here ? I suggest the usual GitFlow which would need to be coordinated with the DevOps team(s) for proper setup of continuous delivery and integration for code coverage and testing.

Should we also setup guides for new members ? It would be nice to have clear documentation of general repository standards for contributions and peer review processes ? 

Also, it might be useful to setup triage strategies for filtering through tasks for labeling / assigning / closing - anyone with experience around automation for this ? 

I might be getting ahead of myself though - thoughts ?

**Review by alexwaddell — DISMISSED** [review]
Could we consider updating #1 on what we will do to say something like ... The team will write good, clean, secure code. ?

I’m a big fan of Martin Fowler and as he so eloquently puts it "Any fool can write code that a computer can understand. Good programmers write code that humans can understand."

**Comment by euan-b** [comment]
Hi. Regarding branching strategy, I agree with Fraser that using short lived (usually 1 dev) feature branches and then merging back to the master should be sufficient. It feels like GitFlow could be overkill here unless we think we’ll need to support multiple versions in the field? Getting the strategy documented would make sense so everyone is clear. GitHub Flow as described here http://scottchacon.com/2011/08/31/github-flow.html sounds fine?

Other stuff: in due course, once tech stacks are chosen, coding style guidelines, maybe rules on testing and more details on reviews/reviewers may be appropriate.

**Review by donal-s — APPROVED** [review]
Looks good

**Review by ai-mindset — APPROVED** [review]
Succinct, friendly and to the point 👍

**Comment by LiamLamb** [comment]
@euan-b I've not tried the suggested github-flow before - I can appreciate what is mentioned around the git-flow and the complexity of it for the less experienced Git users though. I'm personally not too fussed about what strategies we use, but more their simplicity and robustness considering there is likely to be varying levels of experience in code contributions - the suggested sounds fine to me 👍

**Comment by euan-b** [comment]
While the Scott Chacon blog post above, and Vincent Driessen blog post https://nvie.com/posts/a-successful-git-branching-model/ give context around GitFlow and GitHub-flow, a suitable reference for people coming to STA is https://guides.github.com/introduction/flow/

Looks clear, simple and suitable.

**Comment by fedro321** [comment]
> Sounds all good to me :)
> 
> What Git strategies are we wanting to adopt here ? I suggest the usual GitFlow which would need to be coordinated with the DevOps team(s) for proper setup of continuous delivery and integration for code coverage and testing.
> 
> Should we also setup guides for new members ? It would be nice to have clear documentation of general repository standards for contributions and peer review processes ?
> 
> Also, it might be useful to setup triage strategies for filtering through tasks for labeling / assigning / closing - anyone with experience around automation for this ?
> 
> I might be getting ahead of myself though - thoughts ?

Thanks for your input Liam. I've tried not to include any standards in the charter, hopefully the dev's can be pragmatic at review. I've proposed that projects use trunk development and the peer review process is the review you are giving me right now. We block direct pushes to master and have them come in via pull requests with a requirement of one review. We can up that requirement as needed.

The other points are all good thought, but can be addressed per project and not something I want to prescribe.

**Comment by fedro321** [comment]
> Could we consider updating #1 on what we will do to say something like ... The team will write good, clean, secure code. ?
> 
> I’m a big fan of Martin Fowler and as he so eloquently puts it "Any fool can write code that a computer can understand. Good programmers write code that humans can understand."

Hi Alex. I'm also a big fan of Martin Fowler. In my interviews we speak extensively about the SOLID principals and give code tests, we talk about TDD, design patterns, etc. However I feel like this is something different and by demanding such standards we may turn off folks that would otherwise be great contributors. We can always use the code reviews or pair programming to have these discussions during development

**Review by alexwaddell — APPROVED** [review]

**Pull request merged**